src/main/java/com/datadoghq/sketch/ddsketch/store/DenseStore.java

```diff
@@ -235,6 +235,20 @@ public double getTotalCount() {
         return totalCount;
     }
 
+    @Override
+    public void forEach(BinAcceptor acceptor) {
+        for (int i = minIndex; i < maxIndex; i++) {
+            double value = counts[i - offset];
+            if (value != 0) {
+                acceptor.accept(i, value);
+            }
+        }
+        double lastCount = counts[maxIndex - offset];
```

Comment:
I think we actually need to preserve the `minIndex <= maxIndex` check (or a similar check) to skip this if the store is empty.

Reply:
yeah, my bad. Let me add a test which would blow up on this.

Reply:
fixed and added a test for iterating over empty stores